### ISSUE TITLE
Upgrading to v4.9.0 errors

### ISSUE THREAD

**Issue by drizztdourden08** [strapi/strapi#16286]
### Required System information

- Node.js version: 18.12.0
- NPM version: 7.11.2
- Strapi version: 4.9.0
- Database: SQLite
- Operating system: Windows 10
- Is your project Javascript or Typescript: JS

### Describe the bug

I'm not sure why but I wanted to upgrade my project from 4.4.7 to the newest one and got a few bugs right out of the gate. In my original project, it would spit out an error message about the api token salt not being set, but it was in fact. It did that on both 4.7 and 4.9 and couldn't get rid of it.

From there, I backed up to my original version.

I tried to install Strapi in a brand new project and got the following error in a quickstart setup right out of the gate after the install with npx:
Error: Cannot find module 'D:\development\troubleStrapi\my-strapi-project\node_modules\@strapi\icons\dist\index.cjs'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1016:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1009:15)
    at resolveExports (node:internal/modules/cjs/loader:529:14)
    at Module._findPath (node:internal/modules/cjs/loader:569:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:981:27)
    at Function.resolve (node:internal/modules/cjs/helpers:109:19)
    at D:\development\troubleStrapi\my-strapi-project\node_modules\@strapi\admin\webpack.alias.js:41:46
    at Array.reduce (<anonymous>)
    at Object.<anonymous> (D:\development\troubleStrapi\my-strapi-project\node_modules\@strapi\admin\webpack.alias.js:40:22)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)

deleting nodes module and npm i doesn't help here.

To be honest, I have no idea if this is an issue only on my workstation... and only somehow related to the newest versions. I have no clue what the link between the two error message are. On my original project, I have no "Icons" loading error, but get a Salt missing no matter if the salt is indeed set in .env and loading in config admin file using env('API_TOKEN_SALT') like it is by default:
![image](https://user-images.githubusercontent.com/36481532/229609330-12efe48f-4ed1-441e-9877-cfbc33c00c52.png)

Any hint would be appreciated.

**Comment by candidosales** [comment]
@drizztdourden08 , your admin.js should be as the snippet below:

```js
module.exports = ({ env }) => ({
  auth: {
    secret: env("ADMIN_JWT_SECRET", "default_value"),
  },
  apiToken: {
    salt: env("API_TOKEN_SALT", "default_value"),
  },
  transfer: {
    token: {
      salt: env("API_TOKEN_SALT", "default_value"),
    },
  },
});


```

**Comment by derrickmehaffy** [comment]
Which requirements do you mean?

**Comment by drizztdourden08** [comment]
@derrickmehaffy the fact that there is new stuff required in 4.9 that isn’t there is previous version.

When upgrading, it means unless you add these, the application will stop working. These should be either part of the migration documentation or automatically added / triggered through the CLI.

The transfer.token.salt wasn’t part of 4.4 and I spent hours trying to figuring out what’s going on, thing it was the same as the apiToken. I ended up finding it in the doc for admin customization.

something that check the minimal requirement configuration and fix / create the token for you if missing.
